src/engine/blocks.js

```diff
@@ -80,6 +87,34 @@ Blocks.prototype.getOpcode = function (id) {
     return this._blocks[id].opcode;
 };
 
+/**
+ * Get all fields and their values for a block.
+ * @param {?string} id ID of block to query.
+ * @return {!Object} All fields and their values.
+ */
+Blocks.prototype.getFields = function (id) {
+    if (typeof this._blocks[id] === 'undefined') return null;
+    return this._blocks[id].fields;
+};
+
+/**
+ * Get all non-substack inputs for a block.
+ * @param {?string} id ID of block to query.
+ * @return {!Object} All non-substack inputs and their associated blocks.
+ */
+Blocks.prototype.getInputs = function (id) {
+    if (typeof this._blocks[id] === 'undefined') return null;
+    var inputs = {};
+    for (var input in this._blocks[id].inputs) {
+        // Ignore blocks prefixed with substack prefix.
+        if (input.substring(0, Blocks.SUBSTACK_INPUT_PREFIX.length)
+            != Blocks.SUBSTACK_INPUT_PREFIX) {
```

Comment:
One way to do that would be to separate "statement inputs" from "value inputs" during the construction of the block representation:
https://github.com/LLK/scratch-vm/blob/develop/src/engine/adapter.js#L108 i.e., splitting that into two separate cases that get stored separately.

I want to be careful not to complicate our implementation of the "block move" function (https://github.com/LLK/scratch-vm/blob/develop/src/engine/blocks.js#L186) since Blockly treats both types of inputs the same in this case. One way to do that might just to be adding a boolean flag to the input object for "whether it's a statement." Hmm, thoughts?